src/test/java/org/springframework/samples/petclinic/TestDBMethods.java

```diff
@@ -0,0 +1,310 @@
+			"-" + repeatString(n_string, 4) + "-" + repeatString(n_string, 4) +
+			"-" + repeatString(n_string, 12));
+	}
+
+	protected UUID genSimpleUuid(int n) {
+		String n_string = String.valueOf(n);
+		return genSimpleUuid(n_string);
+	}
+
+	protected LocalDate fromStringToTimestamp(String string) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+		return LocalDate.parse(string, formatter);
+	}
+
+	protected void compairSpecialties(Specialty first_specialty, Specialty second_specialty) {
+		try {
+			assertThat(first_specialty.getId()).isEqualTo(second_specialty.getId());
+			assertThat(first_specialty.getName()).isEqualTo(second_specialty.getName());
+		} catch (NullPointerException e) {
+
+		}
+	}
+
+	protected void compairVets(Vet first_vet, Vet second_vet) {
+		assertThat(first_vet.getId()).isEqualTo(second_vet.getId());
+		assertThat(first_vet.getFirstName()).isEqualTo(second_vet.getFirstName());
+		assertThat(first_vet.getLastName()).isEqualTo(second_vet.getLastName());
+	}
+
+	private void comparePetTypes(PetType first_type, PetType second_type) {
+		try {
+			assertThat(first_type.getName()).isEqualTo(second_type.getName());
+			assertThat(first_type.getId()).isEqualTo(second_type.getId());
+		} catch (NullPointerException e) {
+
+		}
+	}
+
+	private void comparePets(Pet first_pet, Pet second_pet) {
+		assertThat(first_pet.getName()).isEqualTo(second_pet.getName());
+		assertThat(first_pet.getId()).isEqualTo(second_pet.getId());
+		assertThat(first_pet.getBirthDate()).isEqualTo(second_pet.getBirthDate());
+		assertThat(first_pet.getBirthDate()).isEqualTo(second_pet.getBirthDate());
+
+		comparePetTypes(first_pet.getType(), second_pet.getType());
+	}
+
+	protected void compairVisits(Visit first_visit, Visit second_visit) {
+		assertThat(first_visit.getId()).isEqualTo(second_visit.getId());
+		assertThat(first_visit.getPetId()).isEqualTo(second_visit.getPetId());
+		assertThat(first_visit.getDate()).isEqualTo(second_visit.getDate());
+		assertThat(first_visit.getDescription()).isEqualTo(second_visit.getDescription());
+	}
+
+	private void compareOwners(Owner first_owner, Owner second_owner) {
+		assertThat(first_owner.getFirstName()).isEqualTo(second_owner.getFirstName());
+		assertThat(first_owner.getLastName()).isEqualTo(second_owner.getLastName());
+		assertThat(first_owner.getId()).isEqualTo(second_owner.getId());
+		assertThat(first_owner.getAddress()).isEqualTo(second_owner.getAddress());
+		assertThat(first_owner.getCity()).isEqualTo(second_owner.getCity());
+		assertThat(first_owner.getTelephone()).isEqualTo(second_owner.getTelephone());
+	}
+
+	private Vet getVetById(UUID id , ArrayList<Vet> vets_array) {
+		int i = 0;
+		for (; i < vets_array.size(); ++i) {
+			if (vets_array.get(i).getId().equals(id)) {
+				break;
+			}
+		}
+
+		return vets_array.get(i);
+	}
+
+	@Test
+	void testVets() {
+		ArrayList<Vet> vets_array = (ArrayList<Vet>) vets.findAll();
```

Comment:
Because you use vetArray more than one plase willbe better convert it to map `Map<UUID, Vet>` converting may be like this
```
list.stream().collect(Collectors.toMap(Vet::getId, Function.identity()))
```